app/controllers/optIn/ConfirmTaxYearController.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.optIn
+
+import auth.{FrontendAuthorisedFunctions, MtdItUser}
+import config.featureswitch.FeatureSwitching
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.ITSAStatusUpdateResponseSuccess
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.optIn.routes.OptInErrorController
+import models.incomeSourceDetails.TaxYear
+import models.optin.ConfirmTaxYearViewModel
+import play.api.Logger
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.DateService
+import services.optIn.OptInService
+import utils.AuthenticatorPredicate
+import views.html.optIn.ConfirmTaxYear
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ConfirmTaxYearController @Inject()(val view: ConfirmTaxYear,
```

Comment:
there is already OptInCompletedController why not update that class with single-year work?

Reply:
or may be renamed it 'OptInCompletedSingleYearController'

Reply:
Okay, sounds like I misunderstood